File: src/main/java/core/utils/CapabilitiesHelper.java
package core.utils;

import org.codehaus.jackson.map.ObjectMapper;

import java.io.File;
import java.io.IOException;
import java.util.HashMap;

public class CapabilitiesHelper {
    public static HashMap<String, String> readAndMakeCapabilities(String fileName) throws IOException {
        String caps = new FileReader().readFile(fileName);
        HashMap<String, String> map = convertCapsToMap(caps);

        updateAbsolutePath(map);
        return map;
    }

    private static HashMap convertCapsToMap(String caps) throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        return mapper.readValue(caps, HashMap.class);
    }

    private static void updateAbsolutePath(HashMap<String, String> map) {
        String path = map.get("app");
        File appPath = new File(path);
        map.put("app", appPath.getAbsolutePath());
    }
}

File: src/main/java/core/utils/CapabilitiesHelper.kt
package core.utils

import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive
import java.io.IOException

object CapabilitiesHelper {
    @Throws(IOException::class)
    fun readAndMakeCapabilities(fileName: String): Map<String, *> {
        val capabilitiesAsJsonString = FileReader().readFile(fileName)
        val json = Json.parseToJsonElement(capabilitiesAsJsonString)
        return json.jsonObject.toMap<String, JsonElement>()
            .mapValues<String, JsonElement, String> { it.value.jsonPrimitive.content }
    }
}
